packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts

```diff
@@ -228,6 +234,33 @@ export class InlineArrayDecoder implements ChunkDecoder {
 	}
 }
 
+/**
+ * Decoder for {@link EncodedIncrementalShape}s.
+ */
+export class IncrementalFieldDecoder implements ChunkDecoder {
+	public constructor(private readonly cache: DecoderContext<EncodedChunkShape>) {}
+	public decode(_: readonly ChunkDecoder[], stream: StreamCursor): TreeChunk {
+		const chunkReferenceIds = readStream(stream);
+		assert(
```

Comment:
We already have an encoder which encodes a sequence of chunks as an array of chunks in some specific format: EncodedNestedArray.

I think the IncrementalChunkDecoder should just read a single chunkReferenceIds.

Doing so reduces complexity, but also makes more optimized compression possible.

For example if we know the encoder will always produce a single chunkReferenceId, the wrapping array can be omitted saving two characters.

Reply:
Updated this as per your suggestion.